core/src/data_sources/data_source.rs

```diff
@@ -934,15 +934,18 @@ impl DataSource {
                                 return Ok(d);
                             }
 
+                            let mut hasher = blake3::Hasher::new();
```

Comment:
For readibility and consistency with the rest of the code base. Lets finalize and assign to `document_id_hash` here

Reply:
Okay